infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -344,13 +347,24 @@ public void initChannel(io.netty.channel.socket.SocketChannel ch) throws Excepti
 
         } catch (Exception ex) {
             log.error("Corfu server shut down unexpectedly due to exception", ex);
+            cleanShutdown();
         } finally {
             bossGroup.shutdownGracefully();
             workerGroup.shutdownGracefully();
         }
 
     }
 
+    /** Attempt to cleanly shutdown all the servers. */
+    public static void cleanShutdown() {
+        log.info("CleanShutdown: Starting Cleanup.");
+        sequencerServer.shutdown();
```

Comment:
What if one of the sequencers throws an exception during shutdown ?

Would't it be better to loop through the servers calling each shutdown with a try/catch ?